cluster-up/cluster/kind/manifests/product-uuid-podpreset.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: settings.k8s.io/v1alpha1
+kind: PodPreset
```

Comment:
should be on e2e tests, as we done it the past, 
see please
https://github.com/kubevirt/kubevirtci/pull/570/commits/7a925f90579a3dd9d9e13078e8a10c461f267486#r596837683

Reply:
I continued this [here](https://github.com/kubevirt/kubevirtci/pull/570#issuecomment-801952386).

Reply:
Lets continue it here please instead of in the root,
kubevirtci should be the one that is agnostic of the e2e tests,
This is what @AlonaKaplan and myself concluded when we worked on migration
and the https://github.com/kubevirt/kubevirt/pull/3374 is a consequence of it.
I find it (mentioned PR above) much better than to have the opposite dependency (kubevirtci to know the NS of the tests),
for example if the test change the NS, or even doesn't want to create a PodPreset,
this way the kubevirtci comes without PodPreset but with support for it, and the user might create it upon need, for his NS.

Reply:
Thank for your input @oshoval.

I think the reasoning you raised is valid but solvable by keeping the tests unaware of this.
You argued in favor of the previous approach but did not feedback to its problems, which seem to be no less important than the one you are raising.

How about this: You make a specific NS migration-ready, and if the relevant tests find this NS, they will use it. Otherwise, they will use the "default" one or just skip the migration tests. If we choose to skip, then we probably need to allow passing the NS name that supports migration through a flag-argument.

Reply:
Honestly i find it over engineering, and prefer that the kubevirtci will be the agnostic product (as expressed before),
i don't see any reason that it will pass stuff in a resolution such as NS.
I less like "scanning" and even less message passing between kubevirtci and kubevirt.

I think that its purely natual that the tests will create a PodPreset in case needed, i just see it otherwise than you,
i don't find it "know what tricks are needed to be done", but its matter of taste i guess.
The tests know that they actually want migration, and for that they create the test-wise layer (PodPreset),
where the kubevirtci does the platform wise prepartion.

This is at least my opinion.

Reply:
What exactly is over-engineered here?
I provided you with an example where this will not makes sense when running the cluster against an external bare-metal node. This "trick" is done only because *you* know that the cluster is kind, but in reality it may not be kind.

The suggestion from my part is to prepare on a *kind* setup a cluster that will support unique UID that allows libvirt to perform migration. It is not dependent on the tests, it just prepares it for anyone that may want to use it.

Reply:
The mentioned PR solved this as well, by creating the PodPreset only if the provider is kind
this also solve the bare metal example i think ?

Up to you, this is my opinion, i prefer agnostic separation, and feels like tprefer a specific NS that is the only one that is migration ready, is too much resolution out of the scope of kubevirtci, and btw also a little limitation,
for example cnv-tests run each tests group in its own namespace,
lets say they would use kubevirtci, and have 2 groups ? what then ?
just an example

Reply:
Its not longer possible to create the PodPrest the way presented on https://github.com/kubevirt/kubevirt/pull/3374
kubevirt uses k8s.io/api v0.20.2 package where we no longer have PodPrest type.

I think we can leave with this for now as it will be valid until the oldest version we would need to support is k8s-1.20

Reply:
We don't want to implement this with an admission webhook so we are prepared for >= 1.20 right ?

Reply:
We do, it is planned, for the meantime we want to use PodPrest workaround at lease so we could test SRIOV migration on CI